core/src/main/java/org/apache/iceberg/stats/StatsCompressionCodec.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+public enum StatsCompressionCodec {
+  /**
+   * LZ4 single compression frame with content size present
+   */
+  LZ4("lz4"),
+
+  /**
+   * Zstandard single compression frame with content size present
+   */
+  ZSTD("zstd"),
+  /**/;
+
+  private final String codecName;
+
+  StatsCompressionCodec(String codecName) {
+    this.codecName = codecName;
+  }
+
+  public String getCodecName() {
+    return codecName;
+  }
+
+  public static StatsCompressionCodec forName(String codecName) {
+    Preconditions.checkNotNull(codecName, "codecName is null");
```

Comment:
we tend to do these like

https://github.com/apache/iceberg/blob/c75ac359c1de6bf9fd4894b40009c5c42d2fee9d/api/src/main/java/org/apache/iceberg/DistributionMode.java#L51-L54

Where we leverage `valueOf` to throw IllegalArgs for us and check for valid enum names

Reply:
https://github.com/apache/iceberg-docs/pull/69 mandates the name to be lowercase.
also, since it's not internal, i wanted to have conversion to and from  file-level name explicit.

Reply:
I agree, we can use the `valueOf` with `toUpperCase`.

Reply:
This will be considered a "reference implementation" of the spec. If we're flexible here, we should be flexible in the spec. Otherwise we would add support for situation that's invalid and should never occur, as if mandating it existence.